lib/std/zig/AstGen.zig

```diff
@@ -1819,18 +1819,15 @@ fn structInitExpr(
     }
 
     switch (ri.rl) {
-        .none => return structInitExprAnon(gz, scope, node, struct_init),
```

Comment:
I really hate to tell you this, but you've fundamentally misunderstood #16865. The proposal explicitly does **not** disallow anonymous struct initializations:
> Untyped struct initializations are still permitted - they are useful for metaprogramming (e.g. std.Build.dependency's args parameter) - but they return a "standard" struct type, with no extra allowed coercions etc.

I'm sorry you've gone to all this effort under a flawed understanding of the proposal. If you'd like to redo this work in the correct way, feel free (please mark this PR as a draft or close it in the meantime) - otherwise, I'm still happy to take it on down the line.

Reply:
Just so I'm not misunderstanding, this example from the issue is still allowed then:
```zig
const anon = .{ .x = 123 };
const S = struct { x: u32 };
const s: S = anon;
_ = s;
```

Reply:
No, that example is a compile error. However, the first line of it is permitted - it just acts as if you created a new, normal, `struct` type for the literal.

Reply:
This change will be a little involved, as it will require constructing struct types quite often in Sema, in a manner similar to reification. It'll also be a bit tricky for incremental compilation due to the new need to track the anonymous init ZIR instructions - now that I think about it, this might not be a great time for this change, since we're largely trying to focus on incremental right now. I'd still happily review and merge a PR implementing it if you do want to work on it, but if there are other things you'd like to work on, it might be better to prioritise those!

Reply:
Right, a struct type just gets created as if you had defined it.

Reply:
Precisely.

Reply:
I'll make an attempt at implementing the proposal as I do want to get a bit more familiar with the compiler frontend